### DISCUSSION TITLE
How to deny warnings?

### DISCUSSION THREAD

**Discussion by mightyiam** [90-008/nix-cargo-integration#175] [Q&A] [answered]
I have something like this:

```nix
      nci.projects.default = {
        path = ./.;
        useClippy = true;
        drvConfig.env.RUSTFLAGS = "--deny warnings";
      };
```

And I artificially introduced a clippy warning. If I run `RUSTFLAGS="--deny warnings" cargo clippy` I can see the warning denied and the command fails.

But `nix flake check` succeeds.

Am I using the wrong option? Is RUSTFLAGS overridden? Is clippy not running? :shrug:

**Comment by 90-008** [comment, marked as answer]
`useClippy` wasn't working properly, i removed it in favour of a similar option to `checkProfile` (https://github.com/yusdacra/nix-cargo-integration/pull/176)

**Reply by mightyiam** [reply]
```nix
        drvConfig.env.RUSTFLAGS = "--deny warnings";
```

Works now! Thank you, @yusdacra!
